library/net.c

```diff
@@ -944,11 +944,16 @@ int dnet_state_micro_init(struct dnet_net_state *st,
 	st->n = n;
 
 	st->la = 1;
-	st->weight = DNET_STATE_DEFAULT_WEIGHT;
 
 	INIT_LIST_HEAD(&st->node_entry);
 	INIT_LIST_HEAD(&st->storage_state_entry);
-	INIT_LIST_HEAD(&st->idc_list);
+	st->idc_root = RB_ROOT;
+	err = pthread_rwlock_init(&st->idc_lock, NULL);
+	if (err) {
+		err = -err;
+		dnet_log_err(n, "Failed to initialize idc_lock lock: err: %d", err);
```

Comment:
`dnet_log_err` uses `errno` to print error message but pthreads doesn't set errno (http://man7.org/linux/man-pages/man7/pthreads.7.html), so there `dnet_log` should be used.